pvp-faq.md

```diff
@@ -291,3 +291,16 @@ conditionally, as you'd need to have access to the `Generic` class and
 be able to define instances for it in the first place, before being
 able to make use of (and thus observe) the default implementation of
 `rnf`.
+
+## Internal modules
+
+There has been a long standing convention in the haskell ecosystem to use
+modules that have a module name prefix/infix/suffix component named `Internal`
+(e.g. `Data.Text.Internal`) and then making these modules exempt from PVP.
```

Comment:
First paragraph should specify what `.Internal` are for (i.e. internal API).

Reply:
Can you be a bit more verbose?

Reply:
The paragraph does not explain *why* `.Internal` modules are used. Suggestion:

> There is custom in the Haskell ecosystem to expose a library internal API via `.Internal` modules
(e.g. `Data.Text.Internal`) and then making these modules exempt from PVP.

My edit also remove the word “convention” which can mean two similar-yet-different things: a “general agreement”  or an “arbitrary custom”.

Reply:
FWIW, `Data.Text.Internal` **is not exempt** from PVP. It's is adhering to PVP and is exporting internals (for advanced usage).

In other words, `Data.Text.Internal` **is a bad example**.

Also the wording is implying that "`Internal` modules are exempt from PVP" practice is widely accepted, but **that is not true** either, as the discussion on #46 has indicated.

The wording here can cause more harm than clarification.

---

I'd try with something like:

> There is a dubious practice of exempting some modules from PVP. Various methods are used:
>
> * Mentioning in the documentation that the module is not under PVP
> * Naming module with prefix/infix/suffix component `Internal`.
> * Combination of above
>
> This practice and any others exemptions from the policy are not permitted by the PVP.
>
> The `Internal` modules are useful to export internals of a library, like in `Data.Text.Internal` in `text`.
>
> If author desires to keep library version churn to a minimum, yet be able to change internals freely,
> then they are adviced to create a separate package for the internal API.
> That approach is outlined  [in Nikita Volvov's blog post "Internal convention is a mistake"](http://nikita-volkov.github.io/internal-convention-is-a-mistake/).

Reply:
I adopted your wording mostly.